recipe/recipe.yaml

```diff
@@ -0,0 +1,212 @@
+      run_exports:
+        - ${{ pin_subpackage("fenics-libdolfinx", upper_bound="x.x.x") }}
+        - if: not win
+          then:
+            - petsc * ${{ scalar }}_*
+        - ${{ mpi | replace("impi", "impi_rt") }}
+    tests:
+      - script:
+          file: test-libdolfinx
+          env:
+            target_platform: ${{ target_platform }}
+        files:
+          recipe:
+            - if: win
+              then: impi-toolchain.cmake
+          source:
+            - cpp/test/
+        requirements:
+          run:
+            - ${{ compiler("c") }}
+            - ${{ compiler("cxx") }}
+            - if: win
+              then: impi-devel
+            - pkg-config
+            - cmake
+            - catch2 3.*
+            - ninja
+            - fenics-ffcx ${{ major_minor }}
+    
+  - package:
+      name: fenics-dolfinx
+    build:
+      script: build-dolfinx
+    requirements:
+      build:
+        - ${{ compiler("c") }}
+        - ${{ stdlib("c") }}
+        - ${{ compiler("cxx") }}
+        - cmake
+        - ninja
+        - pkg-config
+        - if: build_platform != target_platform
+          then: python
+        - if: build_platform != target_platform
+          then: cross-python_${{ target_platform }}
+        - if: build_platform != target_platform
+          then: nanobind
+        - if: mpi == 'openmpi' and build_platform != target_platform
+          then: ${{ mpi }}
+      host:
+        - ${{ mpi | replace("impi", "impi-devel") }}
+        # hdf5 shouldn't be needed here, but solver fails without it
+        - hdf5 ${{ hdf5 }}.* mpi_${{ mpi }}_*
+        - ${{ pin_subpackage("fenics-libdolfinx", exact=True) }}
+        - python
+        - pip
+        - scikit-build-core
+        - wheel
+        - nanobind
+        - fenics-basix ${{ major_minor }}
+        - fenics-basix-nanobind-abi
+        - mpi4py
+        - if: not win
+          then:
+            - petsc * ${{ scalar }}_*
+            - petsc4py
+            - slepc
+            - slepc4py
+        # not actually a dependency of the Python part, but having it here
+        # seems to fix a solver error on libxml2's noicu build
+        - libboost-devel
+      run:
+        # code generation only needs c, not cxx
+        - ${{ compiler("c") }}
+        - pkg-config
+        - python
+        - ${{ pin_subpackage("fenics-libdolfinx", exact=True) }}
+        - cffi
+        - if: not (win or match(python, ">=3.13"))
+          then: cffi <1.17
+        - mpi4py
+        - numpy >=1.21
+        - if: not win
+          then:
+            - petsc4py
+            - slepc4py
+        - fenics-basix ${{ major_minor }}
+        - fenics-ffcx ${{ major_minor }}
+        - fenics-ufl ${{ ufl_version }}
+    tests:
+      - python:
+          pip_check: true
+          imports:
+            - dolfinx
+      - script:
+          file: test-dolfinx
+          env:
+            scalar: ${{ scalar }}
+            target_platform: ${{ target_platform }}
```

Comment:
These shouldn't be required, but are for now to workaround https://github.com/prefix-dev/rattler-build/issues/1317